www/authmemcookie.php

```diff
@@ -91,12 +91,25 @@
 	/* Store the authentication data in the memcache server. */
 
 	$data = '';
+	$amcConfig = SimpleSAML_Configuration::getConfig('authmemcookie.php');
+	$format = strtolower($amcConfig->getBoolean('format', 'implode'));
+	assert('in_array($format, array("implode", "multi", "json"))');
 	foreach($authData as $n => $v) {
-		if(is_array($v)) {
-			$v = implode(':', $v);
+		if(!is_array($v)) {
+			$v = array((string)$v);
+		}
+		if ($format === 'multi') {
```

Comment:
This sounds like a bug. If I understood the comments correctly, _implode_ will yield something like this:

```ini
attribute=value1:value2:value3
```

while _multi_ will yield:

```ini
attribute=value1
attribute=value2
attribute=value3
```

But here we are imploding all the different values with a colon in between in case format is _multi_, which is incoherent with the documentation. So I understand this should be:

```php
if ($format === 'implode') {
```